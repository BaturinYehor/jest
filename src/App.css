.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.wrapper {
height: 100vh;
display: flex;
justify-content: center;
align-items: center;

background: rgba(235,23,71,1);
background: -moz-linear-gradient(top, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
background: -webkit-gradient(left top, left bottom, color-stop(0%, rgba(235,23,71,1)), color-stop(59%, rgba(235,23,71,1)), color-stop(60%, rgba(0,172,238,1)), color-stop(60%, rgba(0,172,238,1)), color-stop(100%, rgba(0,172,238,1)));
background: -webkit-linear-gradient(top, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
background: -o-linear-gradient(top, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
background: -ms-linear-gradient(top, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
background: linear-gradient(to bottom, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#eb1747', endColorstr='#00acee', GradientType=0 );

}

.main {
  height: 60vh;
  background: #e6e6e6;
  box-shadow: 0 0 50px -10px rgba(0,0,0,1.5);
  width: 60%;
  padding: 30px;
  border-radius: 10px;
}

.main {
  height: 80vh;
  background: #e6e6e6;
  box-shadow: 0 0 50px -10px rgba(0,0,0,1.5);
  width: 60%;
  padding: 30px;
  border-radius: 10px;
}

.greeting {
  height: 60vh;
  background: #e6e6e6;
  box-shadow: 0 0 50px -10px rgba(0,0,0,1.5);
  width: 60%;
  padding: 30px;
  border-radius: 10px;
}

.text {
  color: #424242;
  font-family: "Adobe Caslon Pro", "Hoefler Text", Georgia, Garamond, Times, serif;
  letter-spacing:0.1em;
  text-align:center;
  text-transform: lowercase;
  font-size: 14pt;
  font-variant: small-caps;
}

.form {
  height: 80vh;
  padding-top: 100px;
  padding-left: 50px;
  border-top: 2px solid #00ACEE;
  border-bottom: 2px solid #00ACEE;
  border-right: 2px solid #00ACEE;
}

input[type="text"] {
  color: #424242;
  background-color: transparent;
  border: 0;
  border-bottom: solid 2px #EB1747;
  width: 50%;
  padding-bottom: 4px;
  font-weight: lighter;
  margin-bottom: 30px;
  margin-right: 20px;
  font-size: 20px;
  outline: none;
}

.get-name-button {
  position: absolute;
  left:    0;
  bottom:   0;
}

.get-name-form {
  position:relative;
  margin-top: 50px;
  height: 45vh;
}

.img {
  height: 15vh;
  width: 15vh;
  background-blend-mode: multiply;
  background: #e6e6e6 url("nix.png") no-repeat center center;
  background-size: contain;
  position: absolute;
  left:    0;
  bottom:   0;
}

input::-webkit-input-placeholder {
  color: #00ACEE;
}

input::-moz-placeholder {
  text-transform: lowercase;
}

button {
  border: 2px solid #EB1747;
  border-bottom: 4px solid #EB1747;
  padding: 8px 20px;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EB1747;
  color: #fff;
  outline: none!important;
}

button:hover {
  border-bottom: 2px solid #00ACEE;
  position: relative;
  top: 2px;
}

.error {
  color: #EB1747;
  font-weight: bold;
}
