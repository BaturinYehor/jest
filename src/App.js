import React from 'react';
import {Route} from "react-router-dom"
import {BrowserRouter} from 'react-router-dom/cjs/react-router-dom.min';
import GreetingForm from './components/GreetingForm/GeetingForm';
import Main from './components/Main/Main';

class
App extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            name: ''
        }
    };

    render() {
        return (
            <BrowserRouter>
                <Route
                    path='/greeting'
                    component={() => <GreetingForm history={this}/>}
                />
                <Route
                    path='/main'
                    component={() => <Main name={this.state.name}/>}
                />
            </BrowserRouter>
        )
    }
}

export default App