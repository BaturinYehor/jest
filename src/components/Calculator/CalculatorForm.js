import React from "react";

class CalculatorForm extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            redirectToMain: false,
            response: [],
            error: [],
            message: ""
        }
    };

    calculate = (e) => {
        e.preventDefault();

        let x = e.target['x'].value;
        let y = e.target['y'].value;

        if (!isNaN(x) && !isNaN(y)) {
            if (isNaN(x) && isNaN(y)) {
                throw new Error('Not x neither y was set');
            } else {
                throw new Error('Not x neither y was set');
            }
        } else {
            this.setState({message: x + y})
        }
    };

    render() {
        return (
            <div>
                <form onSubmit={this.calculate}>
                    <div className="get-name-form">
                        <div className="row">
                            <div className="board">
                                <div className="text">
                                    fill in the fields, press the button and get the result - a number, if the fields
                                    are
                                    filled with numbers, a string-concatenation of characters, if the fields are filled
                                    with
                                    characters
                                </div>
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-sm-6">
                                <input type="text" name="x" placeholder="x"/>
                            </div>
                            <div className="col-sm-6">
                                <div className="board">
                                    <div className="text">
                                        {this.state.message}
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-sm-6">
                                <input type="text" name="y" placeholder="y"/>
                            </div>
                            <div className="col-sm-6">
                                <div className="get-name-button">
                                    <button>Calculate</button>
                                </div>
                            </div>
                            {this.state.error}
                        </div>
                        <div className="row h-50">
                            <div className="col-sm-9">
                            </div>
                            <div className="col-sm-1">
                                <div className="img"/>
                            </div>
                        </div>
                    </div>
                </form>
            </div>
        )
    };
}

export default CalculatorForm