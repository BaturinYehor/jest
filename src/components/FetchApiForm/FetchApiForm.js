import React from "react";

const API = 'https://961826c1-030a-4d64-9117-ed8352256d12.mock.pstmn.io/info';

class Main extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            response: [],
            error: []
        }
    };

    getInformation = (e) => {
        e.preventDefault();
        this.fetchApi(API);
    };

    fetchApi(api) {
        fetch(api)
            .then(response => {
                if(response.ok) {
                    response.json()
                        .then((data) => {
                        if (data.info) {
                            this.setState({response: data.info});//information fetched from api
                        } else {
                            this.setState({error: data.error.message})//can't find api to fetch
                        }
                    })
                } else {
                    throw new Error('Error while fetching api');
                }
            })
            .catch((error) => {
                this.setState({error: error})//error while fetching information from api
            });
    }

    render() {
        return (
            <div>
                <button onClick={this.getInformation}>
                    What is jest?
                </button>
                <div className="get-api-form">
                    <div className="text">
                        {this.state.response}
                        <div className="error">
                            {this.state.error}
                        </div>
                    </div>
                </div>
            </div>
        )
    };
}

export default Main