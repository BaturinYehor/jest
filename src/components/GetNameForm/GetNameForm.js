import React from "react";

class GetNameForm extends React.Component {

    constructor(props) {
        super(props);
    };

    render() {
        return (
            <div>
                <form onSubmit={this.props.handleSubmit}>
                    <div className="get-name-form">
                        <div className="row">
                            <input type="text" name="name" placeholder="Name"/>
                        </div>
                        <div className="row h-75">
                            <div className="col-sm-10">
                                <div className="get-name-button">
                                    <button>Get information</button>
                                </div>
                            </div>
                            <div className="col-sm-2">
                                <div className="img"/>
                            </div>
                        </div>
                    </div>
                </form>
            </div>
        );
    }
}

export default GetNameForm