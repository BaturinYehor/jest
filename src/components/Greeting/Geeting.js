import React from "react";
import GetNameForm from "../GetNameForm/GetNameForm"

class Greeting extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            redirectToMain: false
        }
    };

    redirectToTarget = () => {
        this.setState({redirectToMain: true})
    };

    getName = (name) => {
        this.props.history.state.name = name;
    };

    isNameFilled() {
        return this.props.history.state.name.length > 0;
    };

    render() {
        if (this.isNameFilled()) {
            this.redirectToTarget();
        }

        return (
            <div>
                <div className="wrapper">
                    <div className="greeting">
                        <div className="text">
                Give us your name and let's get straight to it!
                        </div>
                <GetNameForm getName={this.getName}/>
                    </div>
                </div>
            </div>
        )
    }
}

export default Greeting