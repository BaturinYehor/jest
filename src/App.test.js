import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorForm from './components/Calculator/CalculatorForm'
import GetNameForm from './components/GetNameForm/GetNameForm'
import Greeting from './components/Greeting/Geeting'
import FetchApiForm from './components/FetchApiForm/FetchApiForm'
import App from './App'
import {mount} from 'enzyme';

describe('Test suite', () => {
    it('snapshot renders', () => {
        const component = renderer.create(<CalculatorForm/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Greeting form should have exactly one GetNameForm', () => {
        const wrapper = mount(<Greeting/>);
        expect(wrapper.find(GetNameForm).length).toEqual(1);
    });

    it('passes all props to Counter', () => {
        const wrapper = mount(<Greeting/>);
        const counterWrapper = wrapper.find(GetNameForm);
        expect(counterWrapper.find("GetNameForm").find("button").text()).toEqual('Get information');
    });

    it('Greeting text should be correct', () => {
        const wrapper = mount(<Greeting/>);
        expect(wrapper.find(".text").text()).toEqual('Give us your name and let\'s get straight to it!');
    });

    it('Input field should have correct placeholder', () => {
        const wrapper = mount(<Greeting/>);
        const counterWrapper = wrapper.find(GetNameForm);
        expect(counterWrapper.find("GetNameForm").find("input").props().placeholder).toEqual("Name");
    });

    it('Error should arise', () => {
        const wrapper = mount(<Greeting/>);
        const e = {
            preventDefault: () => {
            },
            target: {
                name: {
                    value: ""
                }
            }
        };
        wrapper.instance().handleSubmit(e);
        expect(wrapper.instance().state.error).toEqual("Name can't be empty");
    });

    it('Name should be set', () => {
        const expectedName = "Nick";

        const appContainer = mount(<App/>);
        const greetingContainer = mount(<Greeting history={appContainer.instance()}/>);
        const e = {
            preventDefault: () => {
            },
            target: {
                name: {
                    value: expectedName
                }
            }
        };
        greetingContainer.instance().handleSubmit(e);
        expect(appContainer.instance().state.name).toEqual(expectedName);
    });

    it('submit handler should be called', () => {
        let submit = jest.fn();
        const wrapper = mount(<GetNameForm handleSubmit={submit}/>);
        wrapper.find('button').simulate('submit');
        expect(submit).toHaveBeenCalledTimes(1);
    });


    it('Fetch should ', () => {
        let submit = jest.fn();
        const wrapper = mount(<GetNameForm handleSubmit={submit}/>);
        wrapper.find('button').simulate('submit');
        expect(submit).toHaveBeenCalledTimes(1);
    });

    it('fetches data from api success', done => {
        const message = "some message";
        const URL = "https://961826c1-030a-4d64-9117-ed8352256d12.mock.pstmn.io/info";
        const mockResponse = {
            info: message
        };
        const e = {
            preventDefault: () => {
            },
        };

        const mockJsonPromise = Promise.resolve(mockResponse);
        const mockFetchPromise = Promise.resolve({
            json: () => mockJsonPromise,
        });
        //jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);
        global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);


        const wrapper = mount(<FetchApiForm/>);
        wrapper.instance().getApiData(e);

        expect(global.fetch).toHaveBeenCalledTimes(1);
        expect(global.fetch).toHaveBeenCalledWith(URL);

        process.nextTick(() => {
            expect(wrapper.state()).toEqual(
                {
                    error: [],
                    response: message
                }
            );

            global.fetch.mockClear();
            delete global.fetch;
            done();
        });
    });
});