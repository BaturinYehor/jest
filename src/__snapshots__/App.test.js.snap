// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite snapshot renders 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="get-name-form"
    >
      <div
        className="row"
      >
        <div
          className="board"
        >
          <div
            className="text"
          >
            fill in the fields, press the button and get the result - a number, if the fields are filled with numbers, a string-concatenation of characters, if the fields are filled with characters
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-6"
        >
          <input
            name="x"
            placeholder="x"
            type="text"
          />
        </div>
        <div
          className="col-sm-6"
        >
          <div
            className="board"
          >
            <div
              className="text"
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-6"
        >
          <input
            name="y"
            placeholder="y"
            type="text"
          />
        </div>
        <div
          className="col-sm-6"
        >
          <div
            className="get-name-button"
          >
            <button>
              Calculate
            </button>
          </div>
        </div>
        <div
          className="error"
        >
          
        </div>
      </div>
      <div
        className="row h-50"
      >
        <div
          className="col-sm-9"
        />
        <div
          className="col-sm-1"
        >
          <div
            className="img"
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
